### ISSUE TITLE
Globby skipps folder with symbols in it's name

### ISSUE THREAD

**Issue by UltraDosaaf** [sindresorhus/globby#81]
Can't make `globby` to search file inside this folder: `C:\Users\me\Desktop\github.com + Globby`.
I've tryed `noext` and other options, but without result.

**Comment by mrmlnc** [comment]
Sorry, @jonschlinkert, but you can take a look at this issue? You have a more experience here :)

```
const mm0 = mm.makeRe('fixtures/github.com + Globby/**/*');

// /^(?:(?:\.[\\\/](?=.))?fixtures[\\\/]github\.com + Globby[\\\/]?\b(?!\.)(?:(?!(?:[\\\/]|^)\.).)*?[\\\/](?!\.)(?=.)[^\\\/]*?(?:[\\\/]|$))$/

const mm1 = mm.makeRe('fixtures/github.com + Globby/**/*', { noext: true});

// /^(?:(?:\.[\\\/](?=.))?fixtures[\\\/]github\.com + Globby[\\\/]?\b(?!\.)(?:(?!(?:[\\\/]|^)\.).)*?[\\\/](?!\.)(?=.)[^\\\/]*?(?:[\\\/]|$))$/

const mm2 = mm.makeRe('fixtures/github.com + Globby/**/*', { noext: false});

// /^(?:(?:\.[\\\/](?=.))?fixtures[\\\/]github\.com + Globby[\\\/]?\b(?!\.)(?:(?!(?:[\\\/]|^)\.).)*?[\\\/](?!\.)(?=.)[^\\\/]*?(?:[\\\/]|$))$/
```

I think that we lost `\` before `+` where `noext` option is disabled.

**Comment by garyking** [comment]
I can confirm that globby won't return any results if passed a dir with square brackets in its name. Anyone got a fix for that?

**Comment by sindresorhus** [comment]
Would you be able to submit a pull request with one or more [failing tests](https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#failing-tests)? That would help getting this fixed faster.

**Comment by vladimiry** [comment]
There is also issue with "+" character, see details here https://github.com/renke/import-sort/issues/71.

**Comment by idevelop** [comment]
Hey folks, having come across this issue while investigating the issue linked above in google datastore library, my take is that this isn't something that globby should fix, since it by design accepts patterns like `+(a|b)`, and it should be the developer's responsibility to properly escape any characters in the path that are unintentional wildcards / patterns.

@sindresorhus do you have any suggestions for how a developer should write a sanitization function that correctly escapes all characters, like `+()[]{}` etc?

**Comment by sindresorhus** [comment]
> do you have any suggestions for how a developer should write a sanitization function that correctly escapes all characters, like `+()[]{}` etc?

I agree that would be useful. Can you open a feature request on https://github.com/mrmlnc/fast-glob? It needs to be implemented there first. Something like `fastGlob.escapeGlob()`.

**Comment by idevelop** [comment]
@sindresorhus done: https://github.com/mrmlnc/fast-glob/issues/158

**Comment by martpie** [comment]
The package `glob-escape` helps developers escaping strings with folder names including characters like `[` `]` `(` `)`, but globby returns an empty array when running `const test= await globby('/some/directory/test (with parenses)/**/*')`.

**Comment by martpie** [comment]
My bad, you have to escape the anti-slash escaping the special caracters as well 🙃 

Ref: https://github.com/micromatch/micromatch/issues/77#issuecomment-257610986

**Comment by thorn0** [comment]
Now that `fast-glob` has `escapePath`, it would be good if `globby` made that function available too.

**Comment by nfantone** [comment]
This seems to have been fixed sometime after opening the issue. I just cloned the repository, installed dependencies and:

```sh
> const globby = require('.');
undefined
> (async () => console.log(await globby('/path/src/pages/jobs/[slug].js')))();
Promise { <pending> }
> [
  '/path/src/pages/jobs/[slug].js'
]
```

**Comment by guiaramos** [comment]
Is there any solution for this?
I have issue when trying to ignore the following:
```bash
pages/[...error]/index.tsx
```

**Comment by vnugent** [comment]
I'm having this issue with parenthesis :

```javascript
  const currentDir = `West side (Left)` // <--- parenthesis

  const climbFiles = await globby([
    `${currentDir}/*.md`,
    `!${currentDir}/index.md`,
  ]);

 // climbFiles = [] empty array

```

**Comment by Kiyozz** [comment]
Same problem with next.js' parenthesis folders and using `xo`

Example structure:
```
src/app/layout.tsx
src/app/(main)/page.tsx
```

`xo --fix src` ← (main)/page.tsx is parsed
`xo --fix "src/app/(main)/page.tsx"` ← is not parsed because the parenthesis being ignored
